tests/unit/App.spec.js

```diff
@@ -44,8 +59,91 @@ describe("App.vue", () => {
 
     // After this line, router is ready
     await router.isReady();
+  });
 
+  it("should render", async () => {
     expect(wrapper.vm.isAuthenticated).toBeTruthy();
-    console.log(wrapper.vm.pending);
+  });
+
+  it("shows dialog box when dialog box is set to be shown", async () => {
+    // dialog box shouldn't be shown at first
+    expect(wrapper.vm.isDialogBoxShown).toBeFalsy();
+    expect(wrapper.find('[data-test="dialogBox"]').exists()).toBeFalsy();
+
+    // set the dialog box to be shown and its properties
+    await store.dispatch("dialog/setDialogTitle", dialogTitle);
+    await store.dispatch("dialog/setDialogDescription", dialogDescription);
+    await store.dispatch("dialog/setConfirmButtonConfig", confirmButtonConfig);
+    await store.dispatch("dialog/setCancelButtonConfig", cancelButtonConfig);
+    await store.dispatch("dialog/setDialogBox");
+
+    // the dialog box should be shown now
+    expect(wrapper.vm.isDialogBoxShown).toBeTruthy();
+    expect(wrapper.find('[data-test="dialogBox"]').exists()).toBeTruthy();
+
+    // all the values for the dialog box should be set appropriately
+    let dialogBoxComponent = wrapper.getComponent({ name: "DialogBox" });
+    expect(dialogBoxComponent.props("title")).toBe(dialogTitle);
+    expect(dialogBoxComponent.props("description")).toBe(dialogDescription);
+    expect(dialogBoxComponent.props("confirmButtonConfig")).toStrictEqual(
+      confirmButtonConfig
+    );
+    expect(dialogBoxComponent.props("cancelButtonConfig")).toStrictEqual(
+      cancelButtonConfig
+    );
+    expect(dialogBoxComponent.props("isCloseButtonShown")).toBeFalsy();
+  });
+
+  describe("dialog box", () => {
+    beforeEach(async () => {
+      await store.dispatch("dialog/setDialogTitle", dialogTitle);
+      await store.dispatch("dialog/setDialogDescription", dialogDescription);
+      await store.dispatch(
+        "dialog/setConfirmButtonConfig",
+        confirmButtonConfig
+      );
+      await store.dispatch("dialog/setCancelButtonConfig", cancelButtonConfig);
+      await store.dispatch("dialog/setDialogBox");
```

Comment:
Why are these `await`? 
None of the actions are actually returning a promise, so no need to await for them